.github/workflows/e2e-tests-workflow.yml

```diff
@@ -0,0 +1,74 @@
+name: E2E tests workflow
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+jobs:
+  tests:
+    name: Run e2e tests
+    strategy:
+      matrix:
+        os: [ubuntu-16.04] # use ubuntu-16.04 as required by cypress: https://github.com/marketplace/actions/cypress-io#important
+        java: [14]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Pull and Run Docker
+        run: |
+          version=1.8.0
+          echo $version
+          if docker pull opendistroforelasticsearch/opendistroforelasticsearch:$version
+          then
+            echo "FROM opendistroforelasticsearch/opendistroforelasticsearch:$version" >> Dockerfile
+            ## The ESRestTest Client uses http by default.
+            ## Need to disable the security plugin to call the rest api over http.
```

Comment:
Is it possible to test with security plugin enabled? No need to answer or change this PR if not clear answer. We can create issue and research later.

Asking this as start detector REST API will start AD job. We'd better test with security plugin enabled to know if AD job can run successfully, generate AD result and show the result correctly on Kibana.

Reply:
That's a good point. I created issue: #209